public/app/features/alerting/unified/components/AlertLabelDropdown.tsx

```diff
@@ -10,19 +11,42 @@ export interface AlertLabelDropdownProps {
   defaultValue?: SelectableValue;
   type: 'key' | 'value';
 }
+const _customFilter = createFilter({ ignoreCase: false });
+function customFilter(opt: SelectableValue, searchQuery: string) {
+  return _customFilter(
+    {
+      label: opt.label ?? '',
+      value: opt.value ?? '',
+      data: {},
+    },
+    searchQuery
+  );
+}
+
+const handleIsValidNewOption = (
+  inputValue: string,
+  value: SelectableValue<string> | null,
+  options: OptionsOrGroups<SelectableValue<string>, GroupBase<SelectableValue<string>>>
+) => {
+  const exactValueExists = options.some((el) => el.label === inputValue);
+  const valueIsNotEmpty = inputValue.trim().length;
+  return !Boolean(exactValueExists) && Boolean(valueIsNotEmpty);
+};
```

Comment:
```suggestion
const handleIsValidNewOption: SelectCommonProps<string>['isValidNewOption'] = (inputValue, value, options) => {
  const exactValueExists = (options as Array<SelectableValue<string>>).some((el) => el.label === inputValue);
  const valueIsNotEmpty = inputValue.trim().length;
  return !Boolean(exactValueExists) && Boolean(valueIsNotEmpty);
};
```

Would this work for your use case? This way you won't need to modify any of the Select's base types.

Reply:
Are these base types correct? They require much casting, so they look to be flawed. That's why this PR changes them

Reply:
@Clarity-89 it works 🙌. Thank you! I have the same question as @konrad147.

Reply:
I actually don't know why the base types are the way they are. I guess this is to allow some other types than `SelectableValues` if needed, however it doesn't seem likely that anything else will be used here, so your solution is fine as well.

Reply:
@Clarity-89 I updated my comment. betterer is complaining

Reply:
thank you for your feedback @Clarity-89 🙌